qiskit/circuit/instructionset.py

```diff
@@ -29,6 +29,7 @@ def __init__(self):
         self.instructions = []
         self.qargs = []
         self.cargs = []
+        self.cregs = cregs
```

Comment:
After talking with @1ucian0 , I think this is the right way of supporting `c_if(int, val)` right now. Can you add a comment here explaining why `cregs` is needed? (Also, maybe consider renaming as `circuit_cregs` or similar.)